aclk/aclk_query.c

```diff
@@ -319,24 +319,18 @@ static char *aclk_encode_response(char *src, size_t content_size, int keep_newli
 #pragma region ACLK_QUERY
 #endif
 
-static usec_t aclk_web_api_request_v1(RRDHOST *host, struct web_client *w, char *url)
+static usec_t aclk_web_api_request_v1(RRDHOST *host, struct web_client *w, char *url, usec_t q_created)
 {
-    usec_t t;
+    usec_t t_db;
+    aclk_metric_mat_update(&aclk_metrics_per_sample.cloud_q_recvd_to_processed, now_realtime_usec() - q_created);
 
-    t = now_monotonic_high_precision_usec();
+    t_db = now_monotonic_high_precision_usec();
     w->response.code = web_client_api_request_v1(host, w, url);
-    t = now_monotonic_high_precision_usec() - t;
+    t_db = now_monotonic_high_precision_usec() - t_db;
 
-    if (aclk_stats_enabled) {
-        ACLK_STATS_LOCK;
-        aclk_metrics_per_sample.cloud_q_process_total += t;
-        aclk_metrics_per_sample.cloud_q_process_count++;
-        if (aclk_metrics_per_sample.cloud_q_process_max < t)
-            aclk_metrics_per_sample.cloud_q_process_max = t;
-        ACLK_STATS_UNLOCK;
-    }
+    aclk_metric_mat_update(&aclk_metrics_per_sample.cloud_q_db_query_time, t_db);
```

Comment:
You are mixing and matching realtime with monotonic time here. The first mat_update uses real time (`now_realtime_usec() - q_created`) and the second one uses monotonic time.

Reply:
the `q_created` is realtime set here `new_query->created = now_realtime_usec();` in fnc `aclk_queue_query`

Reply:
so first `metric_mat_update` compares realtime to realtime and second `metric_mat_update` compares monotonic to monotonic...